cluster-http/src/main/scala/akka/management/cluster/scaladsl/ClusterReadinessCheck.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2017-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.management.cluster.scaladsl
+
+import akka.actor.ActorSystem
+import akka.annotation.InternalApi
+import akka.cluster.{ Cluster, MemberStatus }
+import akka.util.Helpers
+import com.typesafe.config.Config
+
+import scala.collection.JavaConverters._
+import scala.concurrent.Future
+
+/**
+ * Internal API
+ */
+@InternalApi
+private[akka] object ClusterReadinessCheckSettings {
+  def memberStatus(status: String): MemberStatus =
+    Helpers.toRootLowerCase(status) match {
+      case "weaklyup" => MemberStatus.WeaklyUp
+      case "up" => MemberStatus.Up
+      case "exiting" => MemberStatus.Exiting
+      case "down" => MemberStatus.Down
+      case "joining" => MemberStatus.Joining
+      case "leaving" => MemberStatus.Leaving
+      case "removed" => MemberStatus.Removed
+      case invalid =>
+        throw new IllegalArgumentException(
+          s"'$invalid' is not a valid MemberStatus. See reference.conf for valid values"
+        )
+    }
+  def apply(config: Config): ClusterReadinessCheckSettings =
+    new ClusterReadinessCheckSettings(config.getStringList("ready-states").asScala.map(memberStatus).toSet)
+}
+
+final class ClusterReadinessCheckSettings(val readyStates: Set[MemberStatus])
+
+final class ClusterReadinessCheck @InternalApi private[akka] (system: ActorSystem,
```

Comment:
Perhaps rename this to the more specific `ClusterMembershipReadinessCheck`?

Reply:
Yep, much better

Reply:
Went with `ClusterMembershipCheck` as it can be a readiness or liveness check in configuration

Reply:
ok, but it should never be used as a liveness check